sdks/python/apache_beam/io/gcp/experimental/spannerio_write_perf_test.py

```diff
@@ -0,0 +1,146 @@
+* spanner_instance - Spanner Instance ID.
+* spanner_database - Spanner Database ID.
+* input_options - options for Synthetic Source:
+num_records - number of rows to be inserted,
+value_size - the length of a single row,
+key_size - required option, but its value has no meaning.
+
+Example test run on DataflowRunner:
+
+python -m apache_beam.io.gcp.experimental.spannerio_write_perf_test \
+  --test-pipeline-options="
+  --runner=TestDataflowRunner
+  --project='...'
+  --region='...'
+  --temp_location='gs://...'
+  --sdk_location=build/apache-beam.tar.gz
+  --publish_to_big_query=true
+  --metrics_dataset='...'
+  --metrics_table='...'
+  --spanner_instance='...'
+  --spanner_database='...'
+  --input_options='{
+    \"num_records\": 10,
+    \"key_size\": 1,
+    \"value_size\": 1024
+    }'"
+
+This setup will result in a table of 1MB size.
+"""
+
+from __future__ import absolute_import
+
+import logging
+import random
+import uuid
+
+from apache_beam import FlatMap
+from apache_beam import Map
+from apache_beam import ParDo
+from apache_beam.io import Read
+from apache_beam.io.gcp.experimental.spannerio import WriteMutation
+from apache_beam.io.gcp.experimental.spannerio import WriteToSpanner
+from apache_beam.testing.load_tests.load_test import LoadTest
+from apache_beam.testing.load_tests.load_test_metrics_utils import CountMessages
+from apache_beam.testing.load_tests.load_test_metrics_utils import MeasureTime
+from apache_beam.testing.synthetic_pipeline import SyntheticSource
+
+# pylint: disable=wrong-import-order, wrong-import-position
+try:
+  from google.cloud import spanner
+except ImportError:
+  spanner = None
+# pylint: enable=wrong-import-order, wrong-import-position
+
+
+class SpannerWritePerfTest(LoadTest):
+  TEST_DATABASE = None
+
+  def __init__(self):
+    super(SpannerWritePerfTest, self).__init__()
+    self.project = self.pipeline.get_option('project')
+    self.spanner_instance = self.pipeline.get_option('spanner_instance')
+    self.spanner_database = self.pipeline.get_option('spanner_database')
+    self._init_setup()
+
+  def _generate_table_name(self):
+    self.TEST_DATABASE = "{}_{}".format(
+        self.spanner_database, ''.join(random.sample(uuid.uuid4().hex, 4)))
+    return self.TEST_DATABASE
+
+  def _create_database(self):
+    spanner_client = spanner.Client()
+    instance = self._SPANNER_INSTANCE = spanner_client.instance(
+        self.spanner_instance)
+    database = instance.database(
+        self.TEST_DATABASE,
+        ddl_statements=[
+            """CREATE TABLE test (
+            id      STRING(99) NOT NULL,
+            data    BYTES(MAX) NOT NULL
+         ) PRIMARY KEY (id)""",
+        ])
+    database.create()
+
+  def _init_setup(self):
+    """Create database."""
+    self._generate_table_name()
+    self._create_database()
+
+  def test(self):
+    def format_record(record):
+      import base64
+      return base64.b64encode(record[1])
+
+    def make_insert_mutations(element, i):
```

Comment:
Do you mind explain the necessity of feeding the WriteMutation class as an argument instead of directly using it in function?

Reply:
I've tried to import `WriteMutation` directly, I found that the side input functions are thread-safe and beam sdk is not available to it. That's why I passed it as a side input param.

Reply:
I don't think I fully get the problem here, Are you getting some errors if using WriteMutation.insert(...)?

Reply:
Yes, when I try to import WriteMutation in `make_insert_mutations` function, it was giving me an import error. But surprisingly now the import works after rebasing my branch with the master! Updated the code, please review.

Thanks!